src/bcs/serializable/moveStructs.ts

```diff
@@ -276,6 +284,10 @@ export class MoveOption<T extends Serializable & EntryFunctionArgument>
     serializer.serializeBytes(bcsBytes);
   }
 
+  toSimpleValue() {
+    return this.vec.toSimpleValue();
+  }
```

Comment:
You don't want to do some special casing rather than return a vector?

Options are taken in as:
1. value
2. undefined / null

Vs this will get you the vector's value as an array (i'm assuming)

Reply:
Yeah it gives you back [] instead of `undefined` or `null` which I think makes sense because that's what it is on-chain, that's what the node returns for view functions, and ultimately that's how it's represented in the class (it converts everything to a  MoveVector, it just accepts undefined/null as inputs)

Reply:
Also the typing for `undefined` and `null` make it a pain to work with :p

I suppose we can do it, but I figure it's better to keep it consistent/simple

Reply:
We can also change it to be an array of 1 for option on the input, just seemed confusing to me